src/main/java/org/cirdles/topsoil/app/RowFactory.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2015 CIRDLES.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.cirdles.topsoil.app;
+
+import java.util.List;
+import javafx.beans.binding.Bindings;
+import javafx.collections.ObservableList;
+import javafx.event.ActionEvent;
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.MenuItem;
+import javafx.scene.control.TableRow;
+import javafx.scene.control.TableView;
+import javafx.util.Callback;
+import org.cirdles.topsoil.data.entry.Entry;
+
+/**
+ * The row Factory for the TSVTable 
+ * Define the Context Menu
+ * 
+ * @author parizotclement
+ */
+public class RowFactory implements Callback<TableView<Entry>, TableRow<Entry>> {
+    
+    private final TableView<Entry> tableView;
+    private final List<TableRow<Entry>> rows;
+    private final ObservableList<TableRow<Entry>> deselectedRows;
+    
+    public RowFactory(TableView<Entry> tableView, List<TableRow<Entry>> rows, ObservableList<TableRow<Entry>> deselectedRows) {
+        this.tableView = tableView;
+        this.rows = rows;
+        this.deselectedRows = deselectedRows;
+    }
+    
+    @Override
+    public TableRow<Entry> call(TableView<Entry> param) {
+        return setRow();
+    }
+    
+    public final TableRow<Entry> setRow() {
+
+        final TableRow<Entry> row = new TableRow<>();
+        rows.add(row);    //Multiple row selection  
+
+        final ContextMenu rowMenu = new ContextMenu();
+        //Item of the Context Menu
+        MenuItem deselectItem = new MenuItem("Deselect");
+        //Actions of the Context Menu
+        deselectItem.setOnAction((ActionEvent event) -> {
+            this.menuActionEvent(rows, deselectedRows);
+        });
```

Comment:
Why the lambda? You can make a method `handleItemDeselection` that acts as an `ActionEventHandler`. You don't need the parameters for `menuActionEvent` anyways.